### PR TITLE
Scope reduction to enable NULL check to protect dereferencing.

### PR DESCRIPTION
**Pull request by sander-visser**
Signed-off-by: sander-visser <github@visser.se>

## Status
**READY**

## Requires Backporting
NO

## Migrations
NO

## Additional comments
Any additional information that could be of interest

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Changelog updated
- [ ] Backported

## Steps to test or reproduce
Outline the steps to test or reproduce the PR here.

### CODE REVIEW

**Comment by mpg** [comment]
Hi @sander-visser and thanks for your contribution! I've reviewed and approved the change, and it will need to be approved by another team member before we can merge it.

Also, it is our current policy to credit all external contributions in the ChangeLog. Could you add an entry by creating a new file in `ChangeLog.d` (see the Readme in that directory)? Thanks!

File: library/ssl_tls.c [hunk 1]
```diff
@@ -6661,28 +6661,32 @@ int mbedtls_ssl_context_load( mbedtls_ssl_context *context,
...
     {
+#if defined(MBEDTLS_SSL_VARIABLE_BUFFER_LENGTH)
+        size_t out_buf_len = ssl->out_buf_len;
+#else
+        size_t out_buf_len = MBEDTLS_SSL_OUT_BUFFER_LEN;
+#endif
+
         mbedtls_platform_zeroize( ssl->out_buf, out_buf_len );
         mbedtls_free( ssl->out_buf );
         ssl->out_buf = NULL;
```
Comment (irwir) [inline library/ssl_tls.c:6679]: The whole `ssl` record is unconditionally zeroed at the end of the function.
This assignment could be removed.

File: library/ssl_tls.c [hunk 2]
```diff
@@ -6661,28 +6661,32 @@ int mbedtls_ssl_context_load( mbedtls_ssl_context *context,
...
     {
+#if defined(MBEDTLS_SSL_VARIABLE_BUFFER_LENGTH)
+        size_t in_buf_len = ssl->in_buf_len;
+#else
+        size_t in_buf_len = MBEDTLS_SSL_IN_BUFFER_LEN;
+#endif
+
         mbedtls_platform_zeroize( ssl->in_buf, in_buf_len );
         mbedtls_free( ssl->in_buf );
         ssl->in_buf = NULL;
```
Comment (irwir) [inline library/ssl_tls.c:6692]: This assignment could be removed as well.

**Comment by mpg** [comment]
I checked the CI results, and the failure in pr-merge Mbed OS testing was an infrastructure glitch (failing to clone the repo), which is obviously unrelated to this PR and can be disregarded.

File: library/ssl_tls.c (hunk 1 above)
Comment (mpg) [reply library/ssl_tls.c:6679]: Indeed. And removing the assignment would also make things more consistent the other allocated structures (transform, session, etc) that we don't explicitly clear.

However IMO this is quite orthogonal to the goal of this PR, so should probably go in a separate PR.

**Comment by ipodipad** [comment]
Could this patch enhance the bug reported in below?
https://forums.mbed.com/t/memory-leak-and-hardfault-at-ecdh-handshake/6447

**Pull request merged**